diff --git a/coursebuilder/app.yaml b/coursebuilder/app.yaml
index 3f47f88..a5947ed 100644
--- a/coursebuilder/app.yaml
+++ b/coursebuilder/app.yaml
@@ -16,8 +16,14 @@ builtins:
 - deferred: on
 
 libraries:
+- name: django
+  version: "1.4"
 - name: jinja2
   version: "2.6"
+- name: lxml
+  version: "2.3"
+- name: numpy
+  version: "1.6.1"
 - name: webapp2
   version: "2.5.2"
 - name: webob
diff --git a/coursebuilder/appengine_config.py b/coursebuilder/appengine_config.py
index f9c49f9..190818e 100644
--- a/coursebuilder/appengine_config.py
+++ b/coursebuilder/appengine_config.py
@@ -74,6 +74,22 @@ THIRD_PARTY_LIBS = [
     # .zip repackaged from .tar.gz download.
     _Library('python-gflags-2.0.zip', relative_path='python-gflags-2.0'),
     _Library('pyparsing-1.5.7.zip'),
+    _Library('appengine_xblock_runtime'),
+    _Library('XBlock'),
+    _Library(os.path.join('XBlock', 'thumbs')),
+    _Library('cb-xblocks-core'),
+    _Library(os.path.join('edx-platform', 'common', 'lib', 'calc')),
+    _Library(os.path.join('edx-platform', 'common', 'lib', 'capa')),
+    _Library(os.path.join('edx-platform', 'common', 'lib', 'chem')),
+    _Library(os.path.join('edx-platform', 'common', 'lib', 'xmodule')),
+    _Library('nltk-gae.zip'),
+    _Library('BeautifulSoup-3.2.1.zip'),
+    _Library('python-dateutil-2.2.zip'),
+    _Library('Mako-0.9.0.zip'),
+    _Library('MarkupSafe-0.18.zip'),
+    _Library('PyYAML-3.10.zip'),
+    _Library('six-1.5.2.zip'),
+    _Library('capa_stubs'),
 ]
 
 
diff --git a/coursebuilder/main.py b/coursebuilder/main.py
index 5b06f75..6ff1e42 100644
--- a/coursebuilder/main.py
+++ b/coursebuilder/main.py
@@ -39,6 +39,7 @@ from modules.oeditor import oeditor
 from modules.review import review
 from modules.search import search
 from modules.upload import upload
+import modules.xblock_module.xblock_module
 
 # use this flag to control debug only features
 debug = not appengine_config.PRODUCTION_MODE
@@ -57,6 +58,7 @@ oeditor.register_module().enable()
 review.register_module().enable()
 search.register_module().enable()
 upload.register_module().enable()
+modules.xblock_module.xblock_module.register_module().enable()
 
 # register modules that are not enabled by default.
 oauth2.register_module()
